ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/FlatOmitPrefixLayoutExtension.java

```diff
@@ -0,0 +1,111 @@
+ * THE SOFTWARE.
+ */
+package edu.wisc.library.ocfl.core.extension.storage.layout;
+
+import edu.wisc.library.ocfl.core.extension.OcflExtensionConfig;
+import edu.wisc.library.ocfl.core.extension.storage.layout.config.FlatOmitPrefixLayoutConfig;
+
+/**
+ * Implementation of the <a href="https://github.com/OCFL/extensions/blob/c9f2ef4fdb581d2a5070bc37af1ebe54150c998d/docs/NNNN-flat-omit-prefix-storage-layout.md">
+ * Flat Omit Prefix Storage Layout</a> extension.
+ *
+ * @author awoods
+ * @since 2021-06-22
+ */
+public class FlatOmitPrefixLayoutExtension implements OcflStorageLayoutExtension {
+
+    public static final String EXTENSION_NAME = "NNNN-flat-omit-prefix-storage-layout";
+
+    private FlatOmitPrefixLayoutConfig config;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDescription() {
+        return "The OCFL object identifiers are expected to contain prefixes which are removed in the mapping to " +
+                "directory names. The OCFL object identifier prefix is defined as all characters before and including a " +
+                "configurable delimiter.";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void init(OcflExtensionConfig config) {
+        // Only set this.config if it is uninitialized
+        if (this.config == null) {
+
+            // Is arg config null?
+            if (config == null) {
+                throw new IllegalArgumentException("Arg config must not be null!");
+            }
+
+            if (!(config instanceof FlatOmitPrefixLayoutConfig)) {
+                throw new IllegalArgumentException(String.format("This extension only supports %s configuration. Received: %s",
+                        getExtensionConfigClass(), config));
+            }
+
+            FlatOmitPrefixLayoutConfig castConfig = (FlatOmitPrefixLayoutConfig) config;
+
+            validateConfig(castConfig);
+            this.config = castConfig;
+        }
+    }
+
+    private static void validateConfig(FlatOmitPrefixLayoutConfig config) {
+        if (config != null) {
+            String delimiter = config.getDelimiter();
+            if (delimiter == null || delimiter.isBlank()) {
+                throw new RuntimeException("Digest configuration must not be empty!");
+            }
+        }
+    }
+
+    @Override
+    public Class<? extends OcflExtensionConfig> getExtensionConfigClass() {
+        return FlatOmitPrefixLayoutConfig.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapObjectId(String objectId) {
+        if (config == null) {
+            throw new RuntimeException("This extension must be initialized before it can be used.");
+        }
+
+        String[] parts = objectId.split(config.getDelimiter());
+        return parts[parts.length - 1];
```

Comment:
Personally, I think it would be nice if this thew an exception if the mapped directory is equal to `extension` or is empty.

Similar to: https://github.com/UW-Madison-Library/ocfl-java/blob/main/ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/FlatLayoutExtension.java#L84